src/crypto/DeviceList.js

```diff
@@ -431,39 +570,47 @@ class DeviceListUpdateSerialiser {
         // non-null indicates that we have users queued for download.
         this._queuedQueryDeferred = null;
 
-        // sync token to be used for the next query: essentially the
-        // most recent one we know about
-        this._nextSyncToken = null;
+        this._devices = null; // the complete device list
```

Comment:
presumably this ends up being the same object as the `_devices` in DeviceList? is that deliberate? If it is, why not pass it in in the constructor? If it's not, `updateDevicesForUsers` needs to keep a list of old `devices` parameters for updates it's missed.

Reply:
It does, and yeah, I guess it probably does make more sense to just pass it in the constructor. Done.